projects/plugins/protect/src/js/components/scan-page/onboarding-steps.jsx

```diff
@@ -0,0 +1,106 @@
+import { Button, Text, getRedirectUrl } from '@automattic/jetpack-components';
+import { useProductCheckoutWorkflow } from '@automattic/jetpack-connection';
+import { createInterpolateElement } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { JETPACK_SCAN_SLUG } from '../../constants';
+import useAnalyticsTracks from '../../hooks/use-analytics-tracks';
+
+const { adminUrl, siteSuffix, hasRequiredPlan } = window.jetpackProtectInitialState;
+
+const UpgradeButton = props => {
+	const { run } = useProductCheckoutWorkflow( {
+		productSlug: JETPACK_SCAN_SLUG,
+		redirectUrl: adminUrl,
+	} );
+	const { recordEventHandler } = useAnalyticsTracks();
+	const getScan = recordEventHandler( 'jetpack_protect_onboarding_get_scan_link_click', run );
+
+	return <Button variant="link" weight="regular" onClick={ getScan } { ...props } />;
+};
+
+export default [
+	{
+		id: hasRequiredPlan ? 'paid-scan-results' : 'free-scan-results',
```

Comment:
[not blocking] One of the benefits of moving `render_callback` out of this file, and adding context to monitor mounted/rendered step components, was that the conditions for when and where each step is rendered can be left solely to the implementation/UI.

Could we alternatively add two separate objects for "paid-scan-results" and "free-scan-results", and then conditionally render them as applicable for each individually?

This way:
- the UI code defines what is shown on screen based on state;
- the hook controls what the current onboarding state actually is;
- and then this list is dedicated only to defining the content of the steps themselves.

Reply:
Yep! I originally started with that approach but then reconsidered repeating the same code but I can see the tradeoff here of doing that so we can leave the rendering solely to the implementation/UI.